apps/client/app/components/GoodsReceiptForm.tsx

```diff
@@ -0,0 +1,146 @@
+'use client';
+
+import React, { useState } from 'react';
+import { useForm, Controller, useFieldArray } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+import { GoodsReceipt, Item, PurchaseOrder } from '../interfaces';
+
+interface GoodsReceiptFormProps {
+    goodsReceipt?: GoodsReceipt;
+    items: Item[];
+    purchaseOrders: PurchaseOrder[];
+    mode: 'create' | 'edit';
```

Comment:
enum would have been nice here

Reply:
Great suggestion. Refactored to use enum